runtime/nvqir/photonics/PhotonicGates.h

```diff
@@ -0,0 +1,199 @@
+/****************************************************************-*- C++ -*-****
+ * Copyright (c) 2022 - 2024 NVIDIA Corporation & Affiliates.                  *
+ * All rights reserved.                                                        *
+ *                                                                             *
+ * This source code and the accompanying materials are made available under    *
+ * the terms of the Apache License 2.0 which accompanies this distribution.    *
+ ******************************************************************************/
+
+#pragma once
+#include "nvqir/Gates.h"
+
+#include <vector>
+
+namespace nvqir {
+
+/// @brief Enumeration of supported CUDA-Q operations
+enum class PhotonicGateName {
+  PlusGate,
+  BeamSplitterGate,
+  PhaseShiftGate,
+};
+
+/// @brief Returns a precomputed factorial for n up to 30
+double _fast_factorial(int n) {
+  static std::vector<double> FACTORIAL_TABLE = {
+      1.,
+      1.,
+      2.,
+      6.,
+      24.,
+      120.,
+      720.,
+      5040.,
+      40320.,
+      362880.,
+      3628800.,
+      39916800.,
+      479001600.,
+      6227020800.,
+      87178291200.,
+      1307674368000.,
+      20922789888000.,
+      355687428096000.,
+      6402373705728000.,
+      121645100408832000.,
+      2432902008176640000.,
+      51090942171709440000.,
+      1124000727777607680000.,
+      25852016738884976640000.,
+      620448401733239439360000.,
+      15511210043330985984000000.,
+      403291461126605635584000000.,
+      10888869450418352160768000000.,
+      304888344611713860501504000000.,
+      8841761993739701954543616000000.,
+      265252859812191058636308480000000.,
+  };
+  if (n > 30) // We do not expect to get 30 photons in the loop at the same time
+  {
+    throw std::invalid_argument("received invalid value, n <= 30");
+  }
+  return FACTORIAL_TABLE[n];
+}
+
+/// @brief Computes the Kronecker delta of two values
+int _kron(int a, int b) {
+  if (a == b)
+    return 1;
+  else
```

Comment:
don't need the `else` after the return

Reply:
Fixed in 14afd309415d6186111e9b955f7d852d3903d3b5